src/logic/calculate.js

```diff
@@ -0,0 +1,105 @@
+import operate from './operate';
+
+const calculate = (dataObject, buttonName) => {
+  let { total, next, operation } = dataObject;
+
+  const ArithmeticOperators = ['+', '-', 'X', '÷'];
+
+  const numbers = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '.'];
+  const modifierOrPercent = ['+/-', '%'];
+  const modifierOrPercentFunc = (number, operator) => {
+    if (next) {
+      if (operation && !total) {
+        operation = null;
+      }
+      next = operate(next, number, operator);
+    } else if (total) {
+      operation = null;
+      total = operate(total, number, operator);
+    }
+  };
+
+  if (buttonName === 'AC') {
+    total = null;
+    next = null;
+    operation = null;
+  } else if (modifierOrPercent.includes(buttonName)) {
+    if (buttonName === '+/-') {
+      modifierOrPercentFunc(-1, 'X');
+    } else if (buttonName === '%') {
+      modifierOrPercentFunc(100, '÷');
+    }
```

Comment:
Instead of repeating the `modifierOrPercentFunc` function here, why not add the logic to the function?

the number parameter can be `number = buttonName == '%' ? 100 : -1;` inside the `modifierOrPercentFunc` function.



[Optional]
As a side note, `modifierOrPercentFunc` is not a good enough name, I would recommend `calculateModifiers` as both `-/+` and `%` are modifiers.

Reply:
I'll improve on this